extern/filereaderlp/reader.cpp

```diff
@@ -903,8 +903,8 @@ void Reader::readnexttoken(bool& done) {
      this->linebufferpos = 0;
    }
 
-   // if all line has been read and we are at end of file, then stop
-   if (this->linebufferpos == this->linebuffer.size() && this->file.eof()) {

```

Comment:
So, one case this doesn't handle is what happens if "End" is not on its own line. I guess that's no good.